### PR TITLE
[pigeon] chore(readme): deprecated command of running pigeon

### PR DESCRIPTION
**Pull request by EchoEllet**
*Update outdated command in `README.md` of running [pigeon](https://pub.dev/packages/pigeon).*

Same as #7709

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by EchoEllet** [comment]
In #7709 we didn't add a CHANGELOG entry documenting this change (see [this comment](https://github.com/flutter/packages/pull/7709#issuecomment-2380232187)). Let me know if CHANGELOG.md entry is necessary for this change.

**Comment by stuartmorgan** [comment]
> Let me know if CHANGELOG.md entry is necessary for this change.

Please see the first entry in https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#faq

File: packages/pigeon/test/readme_test.dart
Comment (stuartmorgan) [inline packages/pigeon/test/readme_test.dart:1]: README changes don't need a test; the bot comment is just a false-positive due to `lib/generator_tools.dart`.

**Comment by stuartmorgan** [comment]
test-exempt: readme-only change; false positive due to having the package version in the code

**Review by tarrinneal — APPROVED** [review]
Thank you!

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 22.4.2
+
+* Update README.md to use `dart run pigeon` instead of the deprecated command `flutter pub run pigeon`.
```
Comment (stuartmorgan) [inline packages/pigeon/CHANGELOG.md:3]: Updates

(Per the style guide linked in the PR checklist.)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]

**Comment by EchoEllet** [comment]
> Thank you

Thank you, too, for the review. I will ensure to read the full guidelines next time.

**Pull request merged**